R/fortify.anosim.R

```diff
@@ -0,0 +1,49 @@
+##' @title Fortify analysis of similarities (ANOSIM) results
+##' @description Prepares fortified versions of results from \code{\link[vegan]{anosim}} objects.
+##' @details TODO
+##' @param model an object of class \code{\link[vegan]{anosim}}.
+##' @param data original data set. Currently ignored.
+##' @param what character; what aspect of the results should be fortified? \code{"ranks"} returns ranks of dissimilarity entries and names of classies. \code{"stats"} returns significance from permutation  and the value of ANOSIM statistic R.
+##' @param ... additional arguments for other methods. Currently ignored.
+##' @return A data frame. For \code{what = "ranks"}, the data frame contains columns \code{'Rank'} and \code{'Class'} with ranks of dissimilarity. For \code{what = "stats"}, the data frame contains columns \code{'Parameter'} and \code{'Value'} with values permutation test and R statistic.
+##'
+##' @export
+##'
+##' @importFrom ggplot2 fortify
+##'
+##' @author Didzis Elferts
+##'
+##' @examples
+##' data(dune)
+##' data(dune.env)
+##' dune.dist <- vegdist(dune)
+##' attach(dune.env)
+##' dune.ano <- anosim(dune.dist, Management)
+##' 
+##' df <- fortify(dune.ano, what = "ranks")
+##' df.title <- fortify(dune.ano, what = "stats")
+##' 
+##' ggplot(df, aes(x = Class, y = Rank)) +
+##'       geom_boxplot(notch = TRUE, varwidth = TRUE) + 
+##'       labs(subtitle = paste("R = ", df.title$Value[2], ", ", "P = ", df.title$Value[1])) +
+##'       theme(axis.title = element_blank())
+`fortify.anosim` <- function(model, data, 
+                             what = c("ranks","stats"),...) {
```

Comment:
I not sure what is better when handling these extra bits of info. Do we need to get them via a separate option in the `fortify` method? Alternatively these could be stored as attributes on the fortified object - I doubt they are very useful and really `fortify` should be used with `data` to add data to a tidy representation of the data in the original object.

Pulling out summary stats such as this are probably better in a **broom** related package.

Also, `autoplot()` is likely the only place we need these and they can be extracted from the anosim object directly.

Thoughts on this?

Reply:
Yes, now `fortify` produce only data, statistics are used only for plotting.